### PR TITLE
Feat/chat room

### PR DESCRIPTION
**Pull request by MarcusSchweizer91**

### CODE REVIEW

File: backend/src/main/java/com/example/backend/service/ChatService.java [hunk 1]
```diff
@@ -90,4 +88,11 @@ public void afterConnectionClosed(WebSocketSession session, CloseStatus status)
         this.session.remove(session);
     }
 
+    public List<ChatMessage> getPreviousMessages(String senderUsername, String receiverUsername){
+        List<ChatMessage> messages = chatRepo.findAllBySenderUsernameAndReceiverUsername(senderUsername,receiverUsername);
+        messages.addAll(chatRepo.findAllBySenderUsernameAndReceiverUsername(receiverUsername,senderUsername));
```
Comment (VladimirKlippert93) [inline backend/src/main/java/com/example/backend/service/ChatService.java:93]: Not quite sure if there is a better solution for this to get the chat history, but that's exactly how I did it.

**Review by VladimirKlippert93 — APPROVED** [review]

File: backend/src/main/java/com/example/backend/service/ChatService.java (hunk 1 above)
Comment (Flooooooooooorian) [inline backend/src/main/java/com/example/backend/service/ChatService.java:93]: LGTM! Du könntest die Abfragen an deine Datenbank kombinieren mit Oder-Verknüpfungen. `findAllByXOrY`

File: backend/src/main/java/com/example/backend/security/UserController.java [hunk 2]
```diff
@@ -66,6 +66,11 @@ public void removeCardFromFavorites(Principal principal, @PathVariable String ca
         userService.removeCardFromFavorites(cardId, principal.getName());
     }
 
+    @GetMapping("/all")
+    public List<MongoUser> getAllUsers(){
+        return userService.getAllUsers();
+    }
+
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/example/backend/security/UserController.java:73]: Watch out you publish all the users passwords and all other sensetive data here!

File: frontend/src/CardGallery/components/Chat/UserChat.tsx [hunk 3]
```diff
@@ -2,40 +2,66 @@ import React, {useEffect, useState} from "react";
...
     const [messages, setMessages] = useState<ChatMessage[]>([]);
     const [message, setMessage] = useState('');
     const [ws, setWs] = useState<WebSocket | null>(null);
-    const { senderUsername, receiverUsername } = useParams<{ senderUsername: string, receiverUsername: string }>();
+
+    const {receiverUsername} = useParams<{ receiverUsername: string }>();
+
+
 
     console.log(messages);
```
Comment (Flooooooooooorian) [inline frontend/src/CardGallery/components/Chat/UserChat.tsx:23]: Remove console.log

**Pull request merged**